js/bedarf.js

```diff
@@ -1,48 +1,98 @@
-    } else {
-      statusText.textContent = 'Offen'
-      statusText.style.color = 'orange'
+
+
+    if (type === 'available'){
+      let frag = new DocumentFragment()
+      let info = frag.appendChild(document.createElement('div'))
+      info.appendChild(document.createElement('h3')).textContent = `${vm.ipAddress}`
+      info.appendChild(document.createElement('span')).textContent = `Typ: ${vm.virtualMachineType}`
+      info.appendChild(document.createElement('span')).textContent = `Username/Password:`
+      info.appendChild(document.createElement('span')).textContent = `${vm.username}`
+      info.appendChild(document.createElement('span')).textContent = `${vm.initialPassword}`
+
+      frag.appendChild(document.createElement('p')).textContent = vm.description
+
+      let card = makeGenericCard(frag)
+      return card
+
     }
-    frag.appendChild(document.createElement('p')).textContent = vm.description
+    if (type === 'request') {
+      let frag = new DocumentFragment()
+      let info = frag.appendChild(document.createElement('div'))
+      info.appendChild(document.createElement('h3')).textContent = `${vm.employeeEmailAddress}`
+      let vmType = ""
+      if (vm.virtualMachineType === 0){
+        vmType = "WindowsServer2022"
+      } else if(vm.virtualMachineType === 1){
+        vmType = "Windows11"
+      } else if (vm.virtualMachineType === 2){
+        vmType = "Ubuntu2204"
+      }
+      info.appendChild(document.createElement('span')).textContent = `Typ: ${vmType}`
+      info.appendChild(document.createElement('p')).textContent = vm.description
+      let status = frag.appendChild(document.createElement('div'))
+      let statusText = status.appendChild(document.createElement('h3'))
+       if(vm.businessApprovalState === 0){
+        statusText.textContent = 'Genehmigt'
+        statusText.style.color = 'green'
+      } else if(vm.businessApprovalState === 1){
+        statusText.textContent = 'Abgelehnt'
+        statusText.style.color = 'red'
+      }else if (vm.businessApprovalState === 2) {
+         statusText.textContent = 'Offen'
+         statusText.style.color = 'orange'
+       }
+
+      let card = makeGenericCard(frag)
+      return card
+
+    }
+
+
 
-    let card = makeGenericCard(frag)
-    return card
   },
   updateTitle() {
     document.querySelector('#content > h1').textContent = 'Bedarf'
   },
+  showAvailableVMs(){
+    let segment = document.getElementById('available-vms')
+    segment.style.display = 'flex'
+   segment.querySelector('h2').textContent = `Verfügbare VMs`
+  },
+
   hideAvailableDays() {
     document.getElementById('available').style.display = 'none'
   },
   updateNewBtn() {
     document.querySelector('#newBtn a').href = '/new_vm'
   },
-  display(params) {
+  async display(params) {
     // die ganzen andere funktionen hier aufrufen, wie im dbg
     //backend und dann in '#requests' rein
     // sortien nach status? genehmigt -> offen -> abgelehnt
+      this.dbg()

```

Comment:
`dbg = debug` ;)

copy the functionality from *dbg*.
*dbg* should of course not be called in production code...